covigator/database/queries.py

```diff
@@ -107,6 +107,51 @@ def get_lineages(self, source: str) -> List[str]:
             and_(klass.status == JobStatus.FINISHED.name, klass.pangolin_lineage != None)).distinct().order_by(
                 klass.pangolin_lineage.asc()).all()]
 
+    def find_parent_who_label(self, lineage, parent_mapping) -> str:
```

Comment:
Is this called at runtime? I think this could be done on the data loading to avoid a recursive query. It may already be like that, I didn't look enough intoo it